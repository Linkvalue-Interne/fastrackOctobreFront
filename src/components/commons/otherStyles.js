import styled from '@emotion/styled';

// You can use this list to import one or all element below
// eslint-disable-next-line max-len
// { Cursor, Rect, HorizontalFlex, VerticalFlex, Line, Ellipse, BoxHead, BoxGoals, BoxSkills }

export const SkillsSlideContainer = styled.div`
  width: 1000px;
  min-width: 800px;
  margin : 0 auto;

  justify-content: ${(props) => props.justifyContent};
`;

// <HorizontalFlex></HorizontalFlex>
export const HorizontalFlex = styled.div`
  display: flex;
  justify-content: center; 

  margin: ${(props) => props.margin};
  margin-right: ${(props) => props.marginR};
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  position: ${(props) => props.position};
  width: ${(props) => props.width};
`;

// <VerticalFlex></VerticalFlex>
export const VerticalFlex = styled.div`
  display : flex; 
  flex-direction : column;

  position: ${(props) => props.position};
  margin: ${(props) => props.margin};
  margin-top: ${(props) => props.marginTop};
  width: ${(props) => props.width};
  align-items: ${(props) => props.alignItems};
  justify-content: ${(props) => props.justifyContent};
`;

// <BoxHead></BoxHead>
export const BoxHead = styled.div`
  border-bottom: 1px solid grey; 
  width : 80%;
  text-align: center; 
  padding-bottom: 1rem; 
`;

// <BoxGoals></BoxGoals>
export const BoxGoals = styled.div`
  border-bottom: 1px solid grey; 
  width : 80%;
  text-align: center; 
  padding-bottom: 1rem; 
`;

// <BoxSkills></BoxSkills>
export const BoxSkills = styled.div`
  border-bottom: 1px solid grey; 
  width : 80%;
  text-align: center; 
  padding-bottom: 1rem; 
`;

export const Rect = styled.div`
  height: 2rem; 
  width : 4rem;
  border-top: ${(props) => props.BorderTop}px solid #2F3640;
  border-right: ${(props) => props.BorderRight}px solid #2F3640;
  border-bottom: ${(props) => props.BorderBottom}px solid #2F3640;
  border-left: ${(props) => props.BorderLeft}px solid #2F3640;
  border-color: ${(props) => props.theme.colors.black};
  opacity: ${(props) => props.opacity};
  background: ${(props) => props.background};
`;

export const Infos = styled.div`

color: ${(props) => props.theme.colors.white};
position: absolute;


`;
