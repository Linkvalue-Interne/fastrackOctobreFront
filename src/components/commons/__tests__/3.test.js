import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import * as emotion from 'emotion'
import { createSerializer } from 'jest-emotion'
import { BrowserRouter as Router } from 'react-router-dom';
import FiltredButton from '../filterButton';
import { SkillsSlideContainer, Rect, HorizontalFlex, VerticalFlex, BoxHead, BoxGoals, BoxSkills, Infos } from '../otherStyles';


expect.addSnapshotSerializer(createSerializer(emotion));

describe('Rect', () => {
  it('should renders with correct styles', () => {
    const mock = {
      colors: {
        black: '#111',
      },
    };

    const tree = renderer.create(
      <Rect theme={mock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
