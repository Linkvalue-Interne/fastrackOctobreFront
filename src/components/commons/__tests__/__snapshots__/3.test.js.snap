// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rect should renders with correct styles 1`] = `
.emotion-0 {
  height: 2rem;
  width: 4rem;
  border-top: px solid #2F3640;
  border-right: px solid #2F3640;
  border-bottom: px solid #2F3640;
  border-left: px solid #2F3640;
}

<div
  className="emotion-0"
/>
`;
